### ISSUE TITLE
Add a way to filter based on the type of failure

### ISSUE THREAD

**Issue by marco-c** [mozilla/mozci#685]
Follow-up to #679.

Depends on #684.

**Comment by La0** [comment]
We can now do that, as Failure types are available since #684 has been merged.

The mozci configuration (and Taskcluster secret) would support a new list under `autoclassification` named `failure-types`:

```toml
[mozci.autoclassification]
enabled = true
failure-type = ["timeout", "crash"]
```

Only currently supported values are:
- `crash`
- `timeout`
- `generic`

Any other value would raise an exception (invald config).

If a failing test is not from the configured `failure-types`, the `autoclassify` output is set to `False`.

**Comment by La0** [comment]
After discussion with @marco-c, we should set `autoclassify` to `True` only when there is a single failure per task (this is hardcoded)
